### PR TITLE
Init project

### PR DESCRIPTION
**Pull request by Ic3Sandy**

### CODE REVIEW

File: lib/merchant-wallet.ts [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+    const contractProxy = new this.client.Contract(
+      ProxyABI,
+      this.context.getProxy()
+    );
+    this.createWalletTxId = await contractProxy.methods
+      .createProxyWithNonce(
+        this.context.getSingleton(),
+        "0x",
+        getRandomIntAsString()
```
Comment (thebevrishot) [inline lib/merchant-wallet.ts:48]: As I understand we provide only 53 bits security here please bump it to 256 bits to utilize range of salt here

File: lib/merchant-wallet.ts [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+      this.createWalletTxId
+    );
+    if (error) {
+      throw new Error(`Error: ${error.message}`);
+    }
+    if (result.length === 0) {
+      throw new Error("Transaction may not have been mined yet.");
+    }
+    this._address = `0x${
+      result[0].log[0].data.split("000000000000000000000000")[1]
```
Comment (thebevrishot) [inline lib/merchant-wallet.ts:76]: Could you provide an example of `result[0].log[0]` here?

File: lib/merchant-wallet.ts [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+    options?: TxOptions
+  ): Promise<string> {
+    if (owners.length < threshold || owners.length < 1) {
+      throw new Error("Invalid owners or threshold");
+    }
+
+    const contract = new this.client.Contract(SafeABI, await this.address());
+    const txId = await contract.methods
+      .setup(
+        owners,
```
Comment (thebevrishot) [inline lib/merchant-wallet.ts]: Please add comments to descript each field here

**Review by thebevrishot — CHANGES_REQUESTED** [review]

File: lib/merchant-wallet.ts (hunk 1 above)
Comment (Ic3Sandy) [reply lib/merchant-wallet.ts:48]: ok krub

File: lib/merchant-wallet.ts (hunk 3 above)
Comment (Ic3Sandy) [reply lib/merchant-wallet.ts]: ok krub

File: lib/merchant-wallet.ts (hunk 2 above)
Comment (Ic3Sandy) [reply lib/merchant-wallet.ts:76]:
```
  {
    address: '2a633b1780b0d516c4a54436d32ce587e905a182',
    topics: [
      '4f51faf6c4561ff95f067657e43439f0f856d97c04d9ec9070a6199ad418e235'
    ],
    data: '000000000000000000000000f189f2f53fea17fa285f44bfc0f40d1378ead9340000000000000000000000008806c103602fa437248d5b9269fdbd31cfb6501c'
  }
```

Comment (Ic3Sandy) [reply lib/merchant-wallet.ts:76]: `emit ProxyCreation(proxy, _singleton);`

**Review by thebevrishot — APPROVED** [review]
lgtm

**Pull request merged**